posts/Modernize-your-pipeline-to-be-ARM-ready-with-multi-arch-container-builds.md

```diff
@@ -0,0 +1,233 @@
+```
+
+The important fields to notice here are:
+* `spec.containers.images: arm64v8/busybox` - notice the arm specification.
+* `nodeSelector.kubernetes.io/arch: arm64` - this tells kubernetes to run the pod on an arm64 node.
+
+The label `kubernetes.io/arch` is supplied to the nodes when they are added in EKS to the node group. This is specified so that if non-arm nodes are added to the cluster, the pod will still get assigned to the matching node based on the architecture (arch for sure). Authenticate to the cluster and run `kubectl apply -f busybox-arm64.yaml`. This will deploy this pod to a node containing the `arm64` label, as desired.
+
+We could have easily ran this same example with an AMD64 node and specifying our image as `amd64/busybox`. That is because the busybox images is *multi-arch*. If you look at the busybox page on [docker hub](https://hub.docker.com/_/busybox/tags), you should see an `OS/Arch` section under tags. Every version of every image supports multiple architectures. When you pull this image as `busybox:latest` on your local docker daemon, docker will detect your native architecture and pull the correct image.
+
+## Building a multi-arch image and deploying to the EKS cluster
+
+For this section, we will be building a tiny python program. This program, when it runs, prints out the architecture of the machine it is running on. After we code up the program, we will build it inside a gitlab pipeline using docker's buildx. Once built, it will be pushed to a docker registry. Finally, we will deploy it to our kubernetes cluster and see the architecture print out. Please note, additional networking/security steps may be required in order for EKS to pull from a specified container registry.  It's okay if you can't follow along for every step, the important part is understanding the bigger picture. 
+
+In your favorite code editor, write the python program into a file named `print-platform-info.py`:
+```python
+import platform
+print(platform.uname())    
+```
+
+Next, create a `Dockerfile` in the same directory as your python program
+
+```Dockerfile
+FROM python
+COPY ./print-platform-info.py /print-platform-info
+CMD ["python3", "/print-platform-info"]
+```
+Now this is typically where you would scamper off to the command line and run `docker build -t my_image .`, but hold off for just a minute! We have to set docker to use a different *build instance*. In the command line inside the project directory where you have your `Dockerfile` run:
+```bash
+docker buildx create --use
+```
+You may need to add some additional arguments to authenticate to your docker registy in the above command. Check the [docs](https://docs.docker.com/engine/reference/commandline/buildx_use/) for any information or google any errors you may run into. Now it is time to run the actual build command. Back at the command prompt, run:
+```bash
+`docker buildx build --platform linux/arm64,linux/amd64 -t <registry-url>/<image-name>:<image-tag> . --push`
```

Comment:
too many back ticks here since you are already in a bash block you dont need extra